promptops/promptops/skill_export.py

```
@@ -57,42 +63,127 @@ def generate_skill_content(data: Dict[str, Any], raw_data: Dict[str, Any], raw_c
-        if "skill" in tags:
-            import yaml
-            try:
-                raw_content = path.read_text(encoding='utf-8')
-                raw_data = yaml.safe_load(raw_content)
-            except Exception:
-                raw_content = ""
-                raw_data = data
+        try:
+            raw_data = yaml.safe_load(raw_content)
+        except yaml.YAMLError as e:
+            line = getattr(e, 'problem_mark', None) and e.problem_mark.line + 1 or "N/A"
```

Ruff (0.15.15)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)